RELEASE/scripts/autoscend.ash

```diff
@@ -2783,37 +2777,74 @@ boolean LX_getDigitalKey()
 
 int auto_freeCombatsRemaining()
 {
+	return auto_freeCombatsRemaining(false);
+}
+
+int auto_freeCombatsRemaining(boolean log_for_debug)
+{
+
+	void debugPrint(string msg)
+	{
+	  if (!log_for_debug) return;
+	  print(msg, "red");
+	}
+
 	int count = 0;
+	int temp = 0;
```

Comment:
I would remove this and use local variables inside the if statement scope where required as this has the potential to be misleading (and temp isn't used outside of the if statements anyway).

Reply:
I am not sure what you mean here. Do you mean replace
```
**int temp = 0;**
	
	debugPrint("Remaining Free Fights:");
	if(!in_koe() && canChangeToFamiliar($familiar[Machine Elf]))
	{
		temp = 5-get_property("_machineTunnelsAdv").to_int();
		count += temp;
		debugPrint("Machine Elf = " + temp);
	}
	if(snojoFightAvailable())
	{
		temp = 10-get_property("_snojoFreeFights").to_int();
		count += temp;
		debugPrint("Snojo = " + temp);
	}
```
with
```
	debugPrint("Remaining Free Fights:");
	if(!in_koe() && canChangeToFamiliar($familiar[Machine Elf]))
	{
		**int** temp = 5-get_property("_machineTunnelsAdv").to_int();
		count += temp;
		debugPrint("Machine Elf = " + temp);
	}
	if(snojoFightAvailable())
	{
		temp = 10-get_property("_snojoFreeFights").to_int();
		count += temp;
		debugPrint("Snojo = " + temp);
	}
```

Reply:
Almost the second one but you'll need to redeclare temp in each if statement (add the `int` before the first use every time).
You can rename it if you like but since it'll be scoped inside each if it's up to you.

Reply:
change made